app/src/main/java/net/christianbeier/droidvnc_ng/MainService.java

```diff
@@ -68,6 +68,8 @@ public class MainService extends Service {
     private static final int NOTIFICATION_ID = 11;
     final static String ACTION_START = "start";
     final static String ACTION_STOP = "stop";
+    final static String ACTION_START_REPEATER_CONNECTION = "start_repeater_connection";
```

Comment:
API question: Are those intents meant to be sent after `ACTION_START`? If so, how can the caller be sure the Service is up and running when sending `ACTION_START_REPEATER_CONNECTION`? 

If not so, i.e ACTION_START_REPEATER_CONNECTION equal entry point as ACTION_START  you need to wire up all the stuff that's triggered by `ACTION_START`, i.e. permission handling and so on.

Reply:
In my use case, repeater startup `ACTION_START_REPEATER_CONNECTION` always follows server startup `ACTION_START `because I need to update the password, which can only be set at server startup. In this case, the first event occurs at the start of the application, and the second at the request of the users, and thus these events are separated by a large interval. 

To ackknowlege the service startup we could send the "response" intent, that name can be defined in the `ACTION_START `intent as extended data `returnIntent` for example.

Now I see a problem, that after transferring the functionality from ContolService to MainService, VNC server configuration is possible only after the creation MainService, since any intent causes the creation of MainService and the start of the VNC server with the old settings, and only after that it is possible to update the settings in the "ACTION_START" intent 
Do you have any ideas how we can update the settings better?
I see two possibilities:
- move VNC Server creation logic in start intent after saving settings in the preferences
- move remote control back to ControlService (allow to update settings before creating the MainService

Reply:
I just noticed that I didn't implement the last case. It is not possible to create a connection later and with each start the old VNC instance is stopped and started with the new setting. From my point of view, it is also unlikely that someone would need such a use case

Reply:
ok, fixed. Last use case is now supported. I will test  the fix tomorrow

Reply:
tested